aztk/node_scripts/core/config.py

```diff
@@ -28,7 +28,7 @@
 cluster_id = os.environ.get("AZTK_CLUSTER_ID")
 pool_id = os.environ["AZ_BATCH_POOL_ID"]
 node_id = os.environ["AZ_BATCH_NODE_ID"]
-is_dedicated = os.environ["AZ_BATCH_NODE_IS_DEDICATED"]
+is_dedicated = os.environ.get("AZ_BATCH_NODE_IS_DEDICATED") == "true"
```

Comment:
why is this `os.environ.get()` and the above are not?

Reply:
we could switch them all, though those are batch variables so they should always be set

Reply:
[`AZ_BATCH_NODE_IS_DEDICATED` is also a Batch variable though](https://docs.microsoft.com/en-us/azure/batch/batch-compute-node-environment-variables)

Reply:
yeah so should we switch them all to `.get` or leave them with `[]`

Reply:
I feel like they should all be `[]` since that will throw an error if the environment variable is not present. `.get` will not until the variable is used -- and that is more confusing. Setting those environment variables is part of the service contract, so we should be fine failing if the service doesn't set them.